Почему класс AccessDeniedException в нашем случае
лучше наследовать от Exception, а не от RuntimeException?

class AccessDeniedException extends Exception {

}

Что бы клиент обязательно обрабатывал эту ошибку